pyOCD/flash/flash.py

```diff
@@ -105,20 +110,42 @@ def __init__(self, target, flash_algo):
     def minimumProgramLength(self):
         return self.min_program_length
 
-    def init(self):
+    def init(self, adr=None, clk=0, fnc=None):
         """
         Download the flash algorithm in RAM
         """
+        if adr is None:
+            adr = self.getFlashInfo().rom_start 
+        
+        assert fnc in self.INIT_FUNCTIONS
+        
         self.target.halt()
-        self.target.setTargetState("PROGRAM")
+        if self.init_called_num == 0:
+            self.target.setTargetState("PROGRAM")
+        self.init_called_num += 1
 
         # update core register to execute the init subroutine
-        result = self.callFunctionAndWait(self.flash_algo['pc_init'], init=True)
+        result = self.callFunctionAndWait(self.flash_algo['pc_init'], 
+                                          r0=adr, r1=clk, r2=fnc, init=True)
 
         # check the return code
         if result != 0:
             logging.error('init error: %i', result)
 
+    def uninit(self, fnc):
+        """
+        Download the flash algorithm in RAM
+        """
+        assert fnc in self.INIT_FUNCTIONS
```

Comment:
the API is symmetric, why do we pass an operation to uninit. We do not support multi-threaded execution as i can see.